lib/components/form/user-settings.js

```diff
@@ -31,9 +33,16 @@ class UserSettings extends Component {
 
   _enableTracking = () => !this.props.user.trackRecent && this.props.toggleTracking(true)
 
-  _getLocations = (user) => {
-    const locations = [...user.locations]
-    if (!locations.find(l => l.type === 'work')) {
+  _getLocations = () => {
+    const { loggedInUser, user } = this.props
+    const locations = clone(user.locations)
+    const savedLocations = loggedInUser ? clone(loggedInUser.savedLocations) : []
+    // Identify saved locations for non-'forgettability' purposes.
+    savedLocations.forEach(l => { l.origin = 'loggedInUser' })
```

Comment:
I'm confused as to the difference between `savedLocations` and `locations`? Are you merging the middleware persisted locations and the localstorage locations? That's not the goal here: we want to only allow one or the other to be enabled for any given deployment (depending on the config file's persistence strategy).

Reply:
I can make the `user` and `loggedInUser` locations exclusive.

Reply:
@binh-dam-ibigroup, it is probably appropriate to handle this at the reducer level. This component should only know about a generic user's locations. In other words, it should not have to determine by itself which set of locations to choose from, the reducer (and/or startup actions) should handle determining which persistence strategy is defined and set the user object from localstorage/middleware accordingly.

Reply:
> This component should only know about a generic user's locations...

In that case, should the locations be just a prop passed to the component depending on loggedInUser and user?